### PR TITLE
Support for Image Alt input

### PR DESCRIPTION
**Pull request by Shinsina**
Screen Shot 2022-05-24 at 10 29 58 AM
Screen Shot 2022-05-24 at 10 30 12 AM
Screen Shot 2022-05-24 at 10 30 21 AM
Screen Shot 2022-05-24 at 10 30 39 AM
Screen Shot 2022-05-24 at 10 30 42 AM

This will also allow us to prepend something like "Photo of:" to all alt attributes (to account for instances like the remaining violation here)

### CODE REVIEW

**Comment by Shinsina** [comment]
[Accessibility improvements](https://trello.com/c/7OYysXzp/130-accessibility-improvements)

File: services/graphql-server/src/graphql/resolvers/platform/asset.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = {
       const opts = rect.isCropped() ? { ...input.options, rect } : input.options;
       return createSrcFor(host, { fileName, filePath }, opts, {});
     },
-    alt: image => createAltFor(image),
+    alt: (image, { input = {} }) => createAltFor(image, input),
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/asset.js]: `({ ...image, ...input })`

File: packages/image/src/create-alt-for.js [hunk 2]
```diff
@@ -22,7 +22,19 @@ module.exports = ({
   caption,
   name,
   fileName,
-} = {}) => {
+} = {}, { prepend, append } = {}) => {
```
Comment (zarathustra323) [inline packages/image/src/create-alt-for.js]: Move into main params

File: packages/image/src/create-alt-for.js [hunk 3]
```diff
@@ -22,7 +22,19 @@ module.exports = ({
...
+  if (prepend) {
```
Comment (zarathustra323) [inline packages/image/src/create-alt-for.js]: Remove the prepend/append ifs

File: packages/image/src/create-alt-for.js [hunk 4]
```diff
@@ -22,7 +22,19 @@ module.exports = ({
...
+    if (name) return `${prepend} ${clean(altFrom(name))}`;
+    return `${prepend} ${clean(altFrom(fileName))}`;
+  }
+  if (append) {
+    if (displayName) return `${clean(displayName)} ${append}`;
+    if (caption) return `${clean(caption)} ${append}`;
+    if (name) return `${clean(altFrom(name))} ${append}`;
+    return `${clean(altFrom(fileName))} ${append}`;
+  }
   if (displayName) return clean(displayName);
```
Comment (zarathustra323) [inline packages/image/src/create-alt-for.js]: `if (displayName) return clean(displayName, { prepend, append });`

Comment (zarathustra323) [reply packages/image/src/create-alt-for.js]:
```js
const strip = v => htmlEntities.encode(stripHtml(v));

const clean = (v, { prepend, append }) => {
  const values = [];
  if (prepend) values.push(strip(prepend));
  values.push(strip(v));
  if (append) values.push(strip(append));
  return values.join(' ');
};
```

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**